Assets/Scripts/Interactables/Resource/Resource.cs

```diff
@@ -0,0 +1,122 @@
+using Examen.Inventory;
+using Examen.Items;
+using Examen.Networking;
+using Examen.Poolsystem;
+using FishNet.Connection;
+using FishNet.Object;
+using MarkUlrich.Health;
+using UnityEngine;
+
+namespace Examen.Interactables.Resource
+{
+    [RequireComponent(typeof(HealthData))]
+    public class Resource : Interactable
+    {
+        [SerializeField] protected ServerInventory p_serverInventory;
+        [SerializeField] protected Item p_resourceItem;
+        [SerializeField] protected int p_supplyAmount = 1;
+        [SerializeField] protected int p_respawnTime;
+
+        protected HealthData p_healthData;
+        protected bool p_hasHealthData;
+
+        public Item ResourceItem => p_resourceItem;
+
+        private void OnEnable() => RespawnResource();
+
+        private void Start()
+        { 
+            ServerInstance.Instance.OnServerStarted += InitResource; 
+        }
+
+        /// <summary>
+        /// If object isServer, it resurrects this gameobject and calls for the clients to mimic its position and active state.
+        /// </summary>
+        public virtual void InitResource()
+        {
+            if (!IsServer)
+                return;
+
+            if (p_hasHealthData)
+                return;
```

Comment:
No all resources should have health, I did this check here because it was first in OnEnable. but know it just happens when  the server starts so I have deleted it